vclib/src/commonMain/kotlin/at/asitplus/wallet/lib/agent/Holder.kt

```diff
@@ -87,48 +90,61 @@ interface Holder {
-
-    sealed class StoredCredential {
-        data class Vc(
-            val vcSerialized: String,
-            val vc: VerifiableCredentialJws,
-            val status: Validator.RevocationStatus
-        ) : StoredCredential()
-
-        data class SdJwt(
-            val vcSerialized: String,
-            val sdJwt: VerifiableCredentialSdJwt,
-            val status: Validator.RevocationStatus
-        ) : StoredCredential()
-
-        data class Iso(
-            val issuerSigned: IssuerSigned
-        ) : StoredCredential()
+    suspend fun getCredentials(): Collection<StoredCredential>?
+
+    sealed class StoredCredential(
+        open val storeEntry: SubjectCredentialStore.StoreEntry,
+        val status: Validator.RevocationStatus,
+    ) {
+        class Vc(
+            override val storeEntry: SubjectCredentialStore.StoreEntry.Vc,
+            status: Validator.RevocationStatus
+        ) : StoredCredential(
+            storeEntry = storeEntry,
+            status = status
+        )
+
+        class SdJwt(
+            override val storeEntry: SubjectCredentialStore.StoreEntry.SdJwt,
+            status: Validator.RevocationStatus
+        ) : StoredCredential(
+            storeEntry = storeEntry,
+            status = status
+        )
+
+        class Iso(
+            override val storeEntry: SubjectCredentialStore.StoreEntry.Iso,
+            status: Validator.RevocationStatus
+        ) : StoredCredential(
+            storeEntry = storeEntry,
+            status = status
+        )
     }
 
+    data class HolderResponseParameters(
+        val presentationSubmission: PresentationSubmission,
+        val verifiablePresentations: List<CreatePresentationResult>
+    )
+
     /**
-     * Creates a [VerifiablePresentation] serialized as a JWT for all the credentials we have stored,
-     * that match the [credentialSchemes] (if specified). Optionally filters by [requestedClaims] (e.g. in ISO case).
-     *
-     * May return null if no valid credentials (i.e. non-revoked, matching attribute name) are available.
+     * Creates an array of [VerifiablePresentation] and a [PresentationSubmission] to match
+     * the [presentationDefinition].
      */
     suspend fun createPresentation(
         challenge: String,
         audienceId: String,
-        credentialSchemes: Collection<ConstantIndex.CredentialScheme>? = null,
-        requestedClaims: Collection<String>? = null,
-    ): CreatePresentationResult?
+        presentationDefinition: PresentationDefinition,
+        // TODO: add authorization semantics to detect unauthorized requests
+        // - eg. a service provider asking for an attribute he should not be allowed to see
+    ): KmmResult<HolderResponseParameters>
 
     /**
      * Creates a [VerifiablePresentation] with the given [validCredentials].
      *
      * Note: The caller is responsible that only valid credentials are passed to this function!
      */
     // TODO dont make this public
```

Comment:
is this TODO still relevant?

Reply:
I am not sure to be honest...
I think it has already been there before, and I am not aware of its relevance as there is not more information on why the TODO exists in the first place

Reply:
You're right, that's on my side ...